lib/puppet/provider/base_dsc/powershell.rb

```diff
@@ -15,6 +17,49 @@
 Applies DSC Resources by generating a configuration file and applying it.
 EOT
 
+  def initialize(value={})
+    super(value)
+    set_symlink_folder
+  end
+
+  if Puppet::Util::Platform.windows?
+    require 'win32/registry'
+
+    # ensure the symlink folder exists. This also ensures the directory structure
+    # is in place prior to creating the symlink if the folder doesn't exist
+    def set_symlink_folder
+      #TODO:
+      # - Check whether the path contains a symlink #Puppet::FileSystem.symlink?(puppet_modules_vendor_folder)
+      # - To support v3.4.0, not all of these methods existed with Puppet::FileSystem
+      #   Everything was available in 3.5.0 though based
+      #   on https://github.com/puppetlabs/puppet/commit/938b4e7d495f5f15557110040458171e832cecf2
+      puppet_modules_vendor_folder = "#{get_program_files_dir}\\WindowsPowerShell\\Modules\\PuppetVendoredModules"
+      if !Puppet::FileSystem.exist?(puppet_modules_vendor_folder)
+        Puppet::FileSystem.dir_mkpath(puppet_modules_vendor_folder) if !Puppet::FileSystem.dir_exist?(puppet_modules_vendor_folder)
+        Puppet::FileSystem.symlink(vendored_modules_path, puppet_modules_vendor_folder)
+      end
+    end
+
+    # get the absolute program files path, no matter whether the process is
+    # a 32 bit or 64 bit process. Shut off registry redirection and query
+    # the ProgramFilesDir key to retrieve the program files value.
+    def get_program_files_dir
+      program_files_dir = 'C:\Program Files'
```

Comment:
Set you're not using `||= reg['ProgramFilesDir']` below, so I don't think this default is ever used.

Can probably just change line 50 to the following (though `reg['missing_value']` might `throw` anyhow, making that moot)

```ruby
reg['ProgramFilesDir'] || 'C:\\Program Files'
```